src/prettyprint.ml

```diff
@@ -0,0 +1,201 @@
+
+let rec string_of_expr = function
+    Binop(e1, o, e2)	        -> "(" ^
+                                  (string_of_expr e1) ^ " " ^
+                                  (string_of_binop o) ^ " " ^
+							                    (string_of_expr e2)
+                                  ^ ")"
+  | Unop(o, e)                -> (string_of_expr e) ^ (string_of_uop o)
+  | Id(s)                     -> s
+  | Assign(v, e)              -> v ^ " = " ^ string_of_expr e
+  | Access(v, e)              -> v ^ "[" ^ string_of_expr e ^ "]"
+  | Int_Lit(i)			          -> string_of_int i
+  | Double_Lit(f)             -> string_of_float f
+  | Char_Lit(c)               -> Char.escaped c
+  | String_Lit(s)		          -> s
+  | Bool_Lit(true)            -> "true"
+  | Bool_Lit(false)           -> "false"
+  | Unit_Lit(u)               -> "unit"
+  | List_Lit(t, exprs)        -> "list<" ^ (string_of_types t) ^ ">[" ^
+                                  (string_of_exprs exprs) ^ "]"
+  | Set_Lit(t, exprs)         -> "set<" ^ (string_of_types t) ^ ">[" ^
+                                  (string_of_exprs exprs) ^ "]"
+  | Map_Lit(t1, t2, m_exprs)  -> "map<" ^ (string_of_types t1) ^
+                                  ", " ^ (string_of_types t2) ^ ">[" ^
+                                  (string_of_kvs m_exprs) ^ "]"
+  | Tuple_Lit(ts, exprs)      -> "tup<" ^ (String.concat ", "
+                                  (List.map string_of_types ts)) ^ ">(" ^
+                                  (string_of_exprs exprs) ^ ")"
+  | Lambda(lambda)            -> (string_of_lambda lambda)
+  | Call(s, exprs)            -> s ^ "(" ^ string_of_exprs exprs ^ ")"
+  | Actor_comm(m,o,a)         -> "(" ^ (string_of_message m) ^ " " ^
+                                  (string_of_actor_op o) ^ " " ^
+							                    (string_of_actor_type a) ^ ")"
+  | Noexpr                    -> ""
+
+and string_of_exprs exprs =
+  String.concat ", " (List.map string_of_expr exprs)
+
+(* helper to print map key-value pairs *)
+and string_of_kvs kvs =
+  let kv_string kv =
+    let (k, v) = kv in string_of_expr k ^ " -> " ^ string_of_expr v in
+  String.concat ", " (List.map kv_string kvs)
+
+(* Print Statements BRUG*)
+
+and string_of_stmt = function
+    Block(stmts)                  -> "{\n" ^ string_of_stmts stmts ^ "\n}"
+  | Expr(expr)                    -> string_of_expr expr ^ ";"
+  | Return(expr)                  -> "return " ^ string_of_expr expr ^ ";"
+  | Mut(v, t)                     -> "mut " ^ string_of_types t ^ " " ^
+                                      v ^ ";"
+  | Mutdecl(v, t, e)              -> "mut " ^  string_of_types t ^ " " ^ v ^
+                                      string_of_expr e ^";"
+  | Vdecl(v, t, e)                -> string_of_types t ^ " " ^ v ^
+                                      string_of_expr e ^ ";"
+  | If(e, s1, s2)                 -> "if (" ^ string_of_expr e ^ ") {\n" ^
+                                      string_of_stmts s1 ^ "\n} else {\n" ^
```

Comment:
Under the hood, every else has an else. I could add some logic to only print the else if the length of s2 is greater than 0, but I don't think that's worth it